src/goto-symex/symex_builtin_functions.cpp

```diff
@@ -384,7 +390,7 @@ void goto_symext::symex_cpp_new(
              "dynamic_"+count_string+"_value";
   symbol.name="symex_dynamic::"+id2string(symbol.base_name);
   symbol.is_lvalue=true;
-  symbol.mode=ID_cpp;
+  symbol.mode=(code.get(ID_statement)==ID_java_new_array ? ID_java : ID_cpp);
```

Comment:
Are we sure `ID_java_new_array` is the only case for a java statement? I'd rather we wrote something like this to make the code more robust:
```cpp
irep_idt &statement=code.get(ID_statement)
if(statement==ID_java_new_array)
  symbol.mode=ID_java;
else if(statement==ID_cpp_new_array)
  symbol.mode=ID_cpp;
else
  assert(false); // replace with the appropriate invariant